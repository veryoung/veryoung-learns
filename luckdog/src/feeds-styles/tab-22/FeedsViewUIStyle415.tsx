/**
 * 书单文卡片，左图（2图）右文
 * Created by liquid @2019年10月21日21:35:31
 */
import React from 'react';
import FeedsViewItem from '../FeedsViewItem';
import { vectorToArray } from './components/utils';
import { PicPicText, TitleContainer } from './components';
import FeedsProtect from '../../mixins/FeedsProtect';
import { shouldComponentUpdate, countReRender } from '@tencent/luckbox-react-optimize';
import { CommonProps } from '../../entity';

@FeedsProtect.protect
export default class FeedsViewUIStyle415 extends FeedsViewItem<CommonProps> {
  public static getRowType() {
    return 415;
  }

  public constructor(props) {
    super(props);
    this.shouldComponentUpdate = shouldComponentUpdate(this, 'FeedsViewUIStyle415');
  }

  public openUrl(item) {
    super.loadUrl(item.sUrl);
  }

  public render() {
    countReRender(this, 'FeedsViewUIStyle415');
    const { itemBean = {}, index } = this.props;
    const { parsedObject = {} } = itemBean;
    const vData = vectorToArray(parsedObject.vData);

    return (
      <TitleContainer parent={this}>
        {vData.map((item, idx) => (
          <PicPicText
            key={idx}
            cardIndex={index}
            {...item}
            style={{ marginTop: idx && 20 }}
            openUrl={() => this.openUrl(item)}
          />
        ))}
      </TitleContainer>
    );
  }
}
