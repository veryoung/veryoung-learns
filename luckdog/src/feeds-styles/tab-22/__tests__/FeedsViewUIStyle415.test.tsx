import React from 'react';
import renderer from 'react-test-renderer';

import FeedsViewUIStyle415 from '../FeedsViewUIStyle415';
import { commonProps } from '@test/mocks/props';

describe('<FeedsViewUIStyle415/>', () => {
  describe('render testing', () => {
    it('render normal', () => {
      const tree = renderer.create(<FeedsViewUIStyle415 {...commonProps} />);
      expect(tree).toMatchSnapshot();
    });
  });
});
