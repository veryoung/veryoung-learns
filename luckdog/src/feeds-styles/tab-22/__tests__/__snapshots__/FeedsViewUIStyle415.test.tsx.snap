// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedsViewUIStyle415/> render testing render normal 1`] = `
<div
  nativeName="View"
  style={
    Object {
      "marginTop": 8,
    }
  }
>
  <div
    nativeName="View"
    style={
      Object {
        "marginHorizontal": 12,
      }
    }
  />
</div>
`;
